torch_xla/csrc/runtime/debug_macros.h

```diff
@@ -30,6 +37,10 @@
 #define XLA_CHECK_GT(a, b) TF_CHECK_GT(a, b)
 
 template <typename T>
+ABSL_DEPRECATED(
```

Comment:
I don't think we should deprecate this. It has a different use case than GetValueOrThrow().

Sometime we know that the StatusOr must contain a value (e.g. we validated it earlier) and violating this indicates a torch_xla bug. In that case, we do want the failure to crash instead of being reported to the user as a python exception.

Reply:
But, in this case, I think we would better use `ABSL_CHECK(status.ok());`. What do you think?

Reply:
Are you suggesting that instead of
```
a = ConsumeValue(foo);
```
we write
```
ABSL_CHECK(foo.status().ok());
a = foo.value();
```
?

I think the latter is worse:
1. it's more verbose,
2. on failure, it doesn't show the error message in the Status.